src/utilities/s3Storage.ts

```diff
@@ -0,0 +1,48 @@
+import S3, { ClientConfiguration } from "aws-sdk/clients/s3";
+
+export interface S3Credentials extends ClientConfiguration {
+  key: string;
+  secret: string;
+  bucket: string;
+}
+
+export class S3Storage {
```

Comment:
It should implement the StorageInterface I think (although some more stuff might need to change in the storage interface)
```suggestion
export class S3Storage implements StorageInterface {
```